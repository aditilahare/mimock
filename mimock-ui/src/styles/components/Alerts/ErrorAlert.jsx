import React from 'react';
import PropTypes from 'prop-types';
import Alert from '@mui/material/Alert';
import AlertTitle from '@mui/material/AlertTitle';

function ErrorAlert({ title, subTitle, message, dataTestId }) {
	return (
		<Alert
			data-testid={dataTestId}
			severity='error'
			style={{
				marginTop: '2rem',
				width: '90%',
				marginLeft: 'auto',
				marginRight: 'auto',
				padding: '1rem',
			}}
		>
			<AlertTitle>{title}</AlertTitle>
			<strong>{subTitle}</strong>
			<If condition={message && message.length}>
				<span> - </span>
				<span>{message}</span>
			</If>
		</Alert>
	);
}

ErrorAlert.propTypes = {
	title: PropTypes.string.isRequired,
	subTitle: PropTypes.string.isRequired,
	message: PropTypes.string,
	dataTestId: PropTypes.string.isRequired,
};

ErrorAlert.defaultProps = {
	message: '',
};

export default ErrorAlert;
