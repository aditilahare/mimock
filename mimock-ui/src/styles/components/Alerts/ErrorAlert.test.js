import React from 'react';
import { render } from '@testing-library/react';
import ErrorAlert from './ErrorAlert';

describe('ErrorAlert', () => {
	it('renders alert without error message', () => {
		const { container, getByTestId, getByText } = render(
			<ErrorAlert
				title='Error'
				subTitle='Error occurred'
				dataTestId='error-1'
			/>
		);

		expect(getByTestId('error-1')).toBeInTheDocument();
		expect(getByText('Error')).toBeInTheDocument();
		expect(getByText('Error occurred')).toBeInTheDocument();

		expect(container).toMatchSnapshot();
	});

	it('renders alert with error message', () => {
		const { container, getByTestId, getByText } = render(
			<ErrorAlert
				title='Error'
				subTitle='Error occurred'
				dataTestId='error-1'
				message='Server is down'
			/>
		);

		expect(getByTestId('error-1')).toBeInTheDocument();
		expect(getByText('Error')).toBeInTheDocument();
		expect(getByText('Error occurred')).toBeInTheDocument();
		expect(getByText(/Server is down/)).toBeInTheDocument();

		expect(container).toMatchSnapshot();
	});
});
