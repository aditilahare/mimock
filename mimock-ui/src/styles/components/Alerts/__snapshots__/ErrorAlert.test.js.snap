// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorAlert renders alert with error message 1`] = `
<div>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation0 MuiAlert-root MuiAlert-standardError MuiAlert-standard css-xilmwd-MuiPaper-root-MuiAlert-root"
    data-testid="error-1"
    role="alert"
    style="margin-top: 2rem; width: 90%; margin-left: auto; margin-right: auto; padding: 1rem;"
  >
    <div
      class="MuiAlert-icon css-1ytlwq5-MuiAlert-icon"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
        data-testid="ErrorOutlineIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M11 15h2v2h-2zm0-8h2v6h-2zm.99-5C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zM12 20c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
        />
      </svg>
    </div>
    <div
      class="MuiAlert-message css-acap47-MuiAlert-message"
    >
      <div
        class="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom MuiAlertTitle-root css-yslrua-MuiTypography-root-MuiAlertTitle-root"
      >
        Error
      </div>
      <strong>
        Error occurred
      </strong>
      <span>
         - 
      </span>
      <span>
        Server is down
      </span>
    </div>
  </div>
</div>
`;

exports[`ErrorAlert renders alert without error message 1`] = `
<div>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation0 MuiAlert-root MuiAlert-standardError MuiAlert-standard css-xilmwd-MuiPaper-root-MuiAlert-root"
    data-testid="error-1"
    role="alert"
    style="margin-top: 2rem; width: 90%; margin-left: auto; margin-right: auto; padding: 1rem;"
  >
    <div
      class="MuiAlert-icon css-1ytlwq5-MuiAlert-icon"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
        data-testid="ErrorOutlineIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M11 15h2v2h-2zm0-8h2v6h-2zm.99-5C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zM12 20c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
        />
      </svg>
    </div>
    <div
      class="MuiAlert-message css-acap47-MuiAlert-message"
    >
      <div
        class="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom MuiAlertTitle-root css-yslrua-MuiTypography-root-MuiAlertTitle-root"
      >
        Error
      </div>
      <strong>
        Error occurred
      </strong>
    </div>
  </div>
</div>
`;
